content/root/best-practices-guide.textile

```diff
@@ -0,0 +1,97 @@
+---
+title: Best Practices Guide
+section: root
+index: 30
+jump_to:
+  Contents:
+    - Ably 101
+    - Using Ably like a pro
+---
+
+h2(#title). Ably Best Practices Guide
+
+In order to avail the best advantage of our platform, be sure to follow this best practices guide while building your applications using Ably.
+
+h3(#ably-101). Ably 101
+
+Ably Realtime is a cloud-based platform that allows you to easily integrate realtime functionality into your applications. Just plug and play.
+
+h4(#pubsub). Publish and Subscribe
+
+Publish data to a named "channel":/realtime/channels-messages on the Ably Realtime platform, and everyone subscribed to that same channel will receive it.
+
+h4(#notifications). Notify
+
+You can use "Push Notifications":/realtime/push to inform your users about the cool new features you are building in your application.
+
+h4(#presence). Be present
+
+Go online and let others in your application know this by using "Presence":/realtime/presence. You can also see who else is present on a channel even when you yourself are NOT present.
+
+h4(#rest-or-realtime). Remain connected, or not
+
+With the availability of both the "Realtime":/realtime and "REST":/rest libraries, choose what suits your application's needs best and get started.
+
+h4(#auth). Choose the right auth scheme
+
+Use "Basic Authentication":/general/authentication#basic-authentication or "Token Authentication":/general/authentication#token-authentication as well as ensure security by "encrypting your messages.":/realtime/encryption
+
+h3(#using-ably-like-a-pro). Using Ably like a pro
+
+h4. 1. Attaching vs subscribing to a channel
+
+* When your client merely attaches to a channel on Ably, the platform will start sending message data even if your client has not subscribed to the channel. Subscribing to a channel simply registers a listener using which your client can perform additonal operations, however, Ably has not knowledge about this as it only listens for a client to attach to a channel. Bear this in mind while attaching to a channel for publish-only purpose.
```

Comment:
I realise this is largely a copy & paste of my points, but it was not my intention that this would be used in this doc.  This is not a good description of the problem I am afraid.  Think about what we're trying to do here which is help people avoid common gotchas / understand things they may not otherwise obtain from our detailed documentation.  So for example, if this is a checklist of things a user should go through, would something like the following be better:

Title: "Attached channels automatically messages from Ably, even without subscribe listeners"
Body: Once a client has attached to a channel, Ably will automatically start sending messages to that client regardless of whether a subscribe listener has been registered.  A common misunderstanding is that only once a client [subscribes](/realtime/channels-messages#message-subscription) to a channel will Ably start sending messages to the client.  This is not the case as registering a listener with a Channel [subscribe](/realtime/channels-messages#message-subscription) operation, is a client-side operation only and is not communicated to Ably.  The Ably platform, once a channel is attached and if the client has permission to subscribe to messages, will immediately start sending published messages to that client irrespective of whether there are any subscribe listeners registered on that client or not.